app/src/main/java/com/wea/mobileapp/MainActivity.java

```diff
@@ -113,7 +113,7 @@ private View.OnClickListener getMessage() {
 
             List<String> receivedMessages = dbHandler.readCMACS();
             //if no entries are found in the db, the list will contain a single null element
-            if (receivedMessages.get(0) != null) {
+            if (receivedMessages.size() != 0) {
```

Comment:
this change needs to be reverted. receivedMessages will never have size 0, if there are no results index 0 will be null